### PR TITLE
Add cppcheck

### PR DESCRIPTION
**Pull request by A2va**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/c/cppcheck/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package("cppcheck")
+    set_kind("binary")
+
+    set_homepage("https://cppcheck.sourceforge.io/")
+    add_urls("https://github.com/danmar/cppcheck/archive/refs/tags/$(version).tar.gz")
+    add_versions("2.13.0", "8229afe1dddc3ed893248b8a723b428dc221ea014fbc76e6289840857c03d450")
+
+    on_install(function (package)
+        import("package.tools.cmake").install(package, {})
+    end)
```
Comment (waruqi) [inline packages/c/cppcheck/xmake.lua:10]: no on_test

File: packages/c/cppcheck/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package("cppcheck")
+    set_kind("binary")
```
Comment (waruqi) [inline packages/c/cppcheck/xmake.lua:2]: remove it

**Comment by A2va** [comment]
I wonder why on Linux cppcheck is not found. Is the PATH not modified?

**Comment by waruqi** [comment]
maybe cmake install did not install to correct path, you need use os.cp to.install them

**Comment by A2va** [comment]
Cmake install the binary in the correct folder, here the logs from wsl ubuntu
```
Install the project...
-- Install configuration: "Debug"
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/bin/cppcheck
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/__init__.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/cppcheck.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/cppcheckdata.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/findcasts.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/misc.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/misra.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/misra_9.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/naming.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/namingng.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/runaddon.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/threadsafety.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/addons/y2038.py
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/avr.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/bento4.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/boost.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/bsd.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/cairo.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/cppcheck-lib.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/cppunit.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/dpdk.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/embedded_sql.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/emscripten.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/ginac.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/gnu.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/googletest.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/gtk.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/icu.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/kde.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/libcerror.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/libcurl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/libsigc++.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/lua.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/mfc.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/microsoft_atl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/microsoft_sal.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/microsoft_unittest.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/motif.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/nspr.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/ntl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/opencv2.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/opengl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/openmp.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/openssl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/pcre.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/posix.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/python.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/qt.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/ruby.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/sdl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/sfml.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/sqlite3.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/std.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/tinyxml2.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/vcl.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/windows.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/wxsqlite3.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/wxsvg.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/wxwidgets.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/zephyr.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/cfg/zlib.cfg
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/aix_ppc64.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/arm32-wchar_t2.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/arm32-wchar_t4.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/arm64-wchar_t2.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/arm64-wchar_t4.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/avr8.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/cray_sv1.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/elbrus-e1cp.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/mips32.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/msp430_eabi_large_datamodel.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/pic16.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/pic8-enhanced.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/pic8.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/unix32-unsigned.xml
-- Installing: /home/a2va/.xmake/packages/c/cppcheck/2.13.0/6be049d9c586430f8dba83712151ee92/share/Cppcheck/platforms/unix64-unsigned.xml
```

**Pull request merged**